pvlib/pvarray.py

```diff
@@ -0,0 +1,219 @@
+        with relative or absolute efficiencies, and to accommodate data sets
+        which are not perfectly normalized but have a slight bias at
+        the reference conditions. [unitless|%]
+
+    k_d : numeric, negative
+        “Dark irradiance” or diode coefficient which influences the voltage
+        increase with irradiance. [unitless]
+
+    tc_d : numeric
+        Temperature coefficient of the diode coefficient, which indirectly
+        influences voltage. Because it is the only temperature coefficient
+        in the model, its value will also reflect secondary temperature
+        dependencies that are present in the PV module. [unitless]
+
+    k_rs and k_rsh : numeric
+        Series and shunt resistance loss factors. Because of the normalization
+        they can be read as power loss fractions at reference conditions.
+        For example, if k_rs is 0.05, the internal loss assigned to the
+        series resistance has a magnitude equal to 5% of the module output.
+        [unitless]
+
+    Returns
+    -------
+    eta : numeric
+        The efficiency of the module at the specified irradiance and
+        temperature.
+
+    Notes
+    -----
+    The efficiency values may be absolute or relative, and may be expressed
+    as percent or per unit.  This is determined by the efficiency data
+    used to derive values for the 5 model parameters.  The first model
+    parameter k_a is equal to the efficiency at STC and therefore
+    indicates the efficiency scale being used.  k_a can also be changed
+    freely to adjust the scale, or to change the module class to a slightly
+    higher or lower efficiency.
+
+    All arguments may be scalars or array-like. If multiple arguments
+    are array-like they must be the same shape or broadcastable to the
+    same shape.
+
+    See also
+    --------
+    pvlib.pvefficiency.fit_pvefficiency_adr
+
+    References
+    ----------
+    .. [1] A. Driesse and J. S. Stein, "From IEC 61853 power measurements
+       to PV system simulations", Sandia Report No. SAND2020-3877, 2020.
+
+    .. [2] A. Driesse, M. Theristis and J. S. Stein, "A New Photovoltaic Module
+       Efficiency Model for Energy Prediction and Rating," in IEEE Journal
+       of Photovoltaics, vol. 11, no. 2, pp. 527-534, March 2021,
+       doi: 10.1109/JPHOTOV.2020.3045677.
+
+    Examples
+    --------
+    >>> pvefficiency_adr([1000, 200], 25,
+            k_a=100, k_d=-6.0, tc_d=0.02, k_rs=0.05, k_rsh=0.10)
+    array([100.        ,  92.79729308])
+
+    >>> pvefficiency_adr([1000, 200], 25,
+            k_a=1.0, k_d=-6.0, tc_d=0.02, k_rs=0.05, k_rsh=0.10)
+    array([1.        , 0.92797293])
+
+
+    Adapted from https://github.com/adriesse/pvpltools-python
+    Copyright (c) 2022, Anton Driesse, PV Performance Labs
+    All rights reserved.
+    '''
+    k_a = np.array(k_a)
+    k_d = np.array(k_d)
+    tc_d = np.array(tc_d)
+    k_rs = np.array(k_rs)
+    k_rsh = np.array(k_rsh)
```

Comment:
What case is this handling?  Seems like any of the supported input types should broadcast automatically without needing conversion to array first.

Reply:
Lists!